test/html4/test_document.rb

```diff
@@ -24,6 +24,7 @@ def test_exceptions_remove_newlines
           refute_empty(errors, "has errors")
           errors.each do |error|
             assert_equal(error.to_s.chomp, error.to_s)
+            assert_nil(error.path)
```

Comment:
Let's remove this assertion. This is not the right test for it (exercising newlines in error messages) and we have adequate coverage in the other tests.